### PR TITLE
App 29 reusable button component

### PR DESCRIPTION
**Pull request by jacobjdavey**

### CODE REVIEW

File: src/components/ButtonComponent/ButtonComponent.definitions.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// src/components/Button/Button.definitions.ts
+import type { ButtonColors } from '../../constants/constants';
+
+export interface ButtonProps {
+    variant: 'Compact' | 'Regular' | 'Large';
```
Comment (jacoblurie29) [inline src/components/ButtonComponent/ButtonComponent.definitions.ts:5]: Can we make this an enum called `ButtonVariant`?

File: src/components/ButtonComponent/ButtonComponent.module.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+.Black {
+  background-color: black;
+  color: white;
+}
+
+.button:disabled,
+.button.loading {
+  background-color: #cccccc;
+  cursor: not-allowed;
```
Comment (jacoblurie29) [inline src/components/ButtonComponent/ButtonComponent.module.css:61]: Good!!!!!!

File: src/components/ButtonComponent/ButtonComponent.module.css [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.Black {
+  background-color: black;
+  color: white;
+}
+
+.button:disabled,
+.button.loading {
+  background-color: #cccccc;
```
Comment (jacoblurie29) [inline src/components/ButtonComponent/ButtonComponent.module.css:60]: Let's not gray out the button while it is loading. Let's just show the spinner. Can we give the button another property called `disabled` and when that is `true` we can gray out the button?

File: src/components/ButtonComponent/ButtonComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// src/components/Button/Button.tsx
+import styles from './ButtonComponent.module.css';
+// import React from 'react';
+import { MoonLoader } from 'react-spinners';
+import type { ButtonProps } from './ButtonComponent.definitions';
+
+const Button = ({ variant, color, onClick, text, loading }: ButtonProps) => {
+    const buttonClassName = `${styles.button} ${styles[variant]} ${styles[color]}`;
```
Comment (jacoblurie29) [inline src/components/ButtonComponent/ButtonComponent.tsx:8]: This is awesome.

**Review by mahdmohsin54 — APPROVED** [review]

File: src/components/ButtonComponent/ButtonComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+    return (
+        <button className={buttonClassName} onClick={onClick} disabled={loading}>
+            {loading ? <MoonLoader size={30} color="#FFFFFF" /> : text}
```
Comment (jacoblurie29) [inline src/components/ButtonComponent/ButtonComponent.tsx:12]: It appears the `MoonLoader` does not show properly. The button should look like this:

#### Loading
Screen Shot 2023-11-19 at 9 10 07 PM

#### Icon
Screen Shot 2023-11-19 at 9 08 45 PM

#### No Icon
Screen Shot 2023-11-19 at 9 11 14 PM

File: src/components/ButtonComponent/tests/ButtonComponent.test.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    test("Test Button component to render", () => {
+        const component = render(<ButtonComponent
+            variant="Regular"
+            color={ButtonColors.Blue}
+            onClick={handleButtonClick}
+            text="Test Button"
+            loading={true}
+          />);
+      // Check that the component renders
+      expect(component).toBeTruthy();
```
Comment (jacoblurie29) [inline src/components/ButtonComponent/tests/ButtonComponent.test.tsx:20]: Great test! Can we check for the text?

**Review by jacoblurie29 — CHANGES_REQUESTED** [review]
Great job! Let's work on fixing that spinner and getting the styling to look like the screen shots that I attached. Use the font `Inter` for now

**Pull request merged**